crates/oxc_prettier/src/printer.rs

```diff
@@ -58,23 +61,73 @@ impl<'a> Printer<'a> {
     /// * <https://github.com/prettier/prettier/blob/0176a33db442e498fdb577784deaa77d7c9ae723/src/document/printer.js#L302>
     fn print_doc_to_string(self) -> String {
         let mut out = vec![];
-        let mut cmds: Vec<Command> = vec![Command::new(Indent::root(), Mode::Break, self.doc)];
+        let mut cmds: Vec<Command> = vec![Command::new(Indent::root(), Mode::Break, &self.doc)];
+
+        let mut pos: u16 = 0;
 
         while let Some(Command { indent, doc, mode }) = cmds.pop() {
-            match doc {
+            match &doc {
                 Doc::Str(string) => {
                     out.extend(string.as_bytes());
+                    pos += string.len() as u16;
                 }
                 Doc::Array(docs) => {
                     cmds.extend(docs.into_iter().rev().map(|doc| Command::new(indent, mode, doc)));
                 }
                 Doc::Indent(docs) => {
-                    cmds.extend(docs.into_iter().rev().map(|doc| Command::new(indent, mode, doc)));
+                    cmds.extend(docs.into_iter().rev().map(|doc| {
+                        Command::new(Indent { value: " ", length: indent.length + 1 }, mode, doc)
+                    }));
                 }
                 Doc::Group(docs) => {
-                    cmds.extend(docs.into_iter().rev().map(|doc| Command::new(indent, mode, doc)));
+                    match mode {
+                        Mode::Flat => {
+                            // TODO: consider supporting `group mode` e.g. Break/Flat
+                            cmds.extend(
+                                docs.into_iter()
+                                    .rev()
+                                    .map(|doc| Command::new(indent, Mode::Flat, doc)),
+                            );
+                        }
+                        Mode::Break => {
+                            #[allow(clippy::cast_possible_wrap)]
+                            let remaining_width = (self.options.print_width - pos) as isize;
```

Comment:
this _has_ to be an `isize` to avoid overflow if the pos > print width.

It could be something else. (`i16`?) but must be signed - not sure